tests/Unit/Encoder/ParseChromiumBookmarksTest.php

```diff
@@ -0,0 +1,349 @@
+            'url'         => 'https://framasoft.org/',
+            'tags'        => [
+                'self-hosting',
+            ],
+            'description' => null,
+            'dateCreated' => 1466011661,
+            'public'      => null,
+        ],
+        [
+            'name'        => 'GitHub - shaarli/Shaarli: The personal, minimalist, super-fast, database free, bookmarking service - community repo',
+            'image'       => 'data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAACJElEQVQ4jY1TMWsUQRT+5r3d3Cbe7t3u3hEDdoJgIVieGo1YBixsBIsUtvkHNpaCnaJFUMEihSlEbGxFE8XCRrDWIAqJyd3t3JGcuduZeRa5DUtAk6968/i+733zhlE4hHq9foWZbwG4pkRmAECU2gDw1lr7Qmu9WuarUs2NOH7Onrdw2LQMa8xyO8tuA7BlA07TdM0juvA/cQHj3KdOp3MZgGUAaMTxssc8nxuzOsrzm90s2yDmU6TUtoh0BKhut9t3Rnl+1/O8Mz7z7GSlcnqwt/cKURS1pptNmW42JUmSpfGQKoCp0tBqkTZJkqWCH0VRiyaYFw/uZ+2zcbkDYFAy2AEgAOCcWymaE8yLBOZZALDOjQB0jrGCdevccH9zPEsYPxWU2uz1eutHqbXWP6DUbwCAyAwV0QDUANAxEhBE6uNaSICfAMBK1dI0vX6UOo7jeSaKsD/5F5FSa04EW+22dca8jMPw0r/ESRheZKKnxVlE3lFuzOPRMIeIWN3v93u7ux+IaBBFUasgBkFwlYi2/SD4yEQnSwaPSGv9xa/4T6abzYm6789Vq9WFMAw/K6W2CmKlUtmJoqhRTmNFHmRZ9vXgLyRx/J6IzlnnzotIX2utS/ywkaabTDQFANaY1+0su7G/0TG6WTYHpVaY6LvH/A3ATMngRCE2xtwrxADA5ViDweDNZBDsQamG7/vLw+HwDwDUajVfiZx1wMNOt3u/rPkLJe7aBdfH1TYAAAAASUVORK5CYII=',
+            'url'         => 'https://github.com/shaarli/Shaarli',
+            'tags'        => [
+                'self-hosting',
+            ],
+            'description' => null,
+            'dateCreated' => 1466012934,
+            'public'      => null,
+        ],
+        [
+            'name'        => 'ownCloud.org',
+            'image'       => 'data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAAC2ElEQVQ4jW2TS2hdVRSGv7XPud7cvBrbmMQ0NyZtIokltVgbFUuiNgUtGWhBKtiOHAh2kI50VOhA6NSJOBacxEkmHRTFFDKoz0pbk9IohYrEPBriI7nnpvfcs34H96Y+cLNhb/Ze++ffa63PeobHBi3KfSizl5ACgJkZ/zMkiVqAmzSrLD0btXQNTIcod1xy1d9aXYQQApKQhJn9LWymEOUGHIatZ2SiWpcPZmYhBCFZUt4mrVZpLBSI4wh3B5CZmWqKDhAbRFLtwt21uVUyM2P0qREeK3Yzd/Vbfv9jk0JDnmrmJkl1JwGwuHaAuUtNjQU7Nv4s3Z2PcOG9d8jnH2Lh9k+8euYcK6vrtO1qYcfBzhq1duy7EMcRpVJi488f4eL5KQb29dLZsYfl1XXy+Tz9vd08MbSfG/OL1LNr9bxYHEJge7vC/UpKX7Gbza2EO3d/oVy+T/vuNippyuD+Pt58fZKN3/7kg48+ptjTjZkhiThJyvT17uXQyBDvnnuL1uYm1u5tcPDAIIWGBgDStMrs3NecOD7Gw22tfPLpJZJymTiO4OiJM1peuadbi3dUqVS0tr6ha9cXJElpmipNU7m7bswvqpQkkqTpmcsq7H1aXcMvKJycnKCrs51cLuaba/Nc/+E2O+V2F+56YHerVKZSSTk2/gxTb5+m+GgX8c2FHwEY6O9laXmVF4+O8uvKGkl5m8ZC7QtJeZssczradwOwfGedi+eneL+xAet4fMxPvfayPXfkSc1d/c5GDx/k+5u3uPvzEm+cfAUB0zOXaSoUOH1qkqXlVT6b/VKHDx2wmUufy4ojE9oqJaqXRoDl4pjMHUkgCFEAIAQjy1xmWKVSVVNjwWJB1trSjOqtXGfmPzzpwTTDJNTSHDxzJ0i6IkLkkmdZJnc3SWRZhrvj7mSZ4zWozN3lkrsscvcr0a49xa+AIcz6gR0S/+VgZ/8PnIVnX5ClZ/8Ce+CJ7pbOimUAAAAASUVORK5CYII=',
+            'url'         => 'https://owncloud.org/',
+            'tags'        => [
+                'self-hosting',
+            ],
+            'description' => null,
+            'dateCreated' => 1466013448,
+            'public'      => null,
+        ],
+    ];
```

Comment:
This part must have been real fun :scream_cat:

Reply:
I had to write a small `SublimeText` python custom plugin to convert console `var_dump()` into proper php arrays... That was actually quite fun. Amazing what can be done with regular expressions.